otelcol/command.go

```diff
@@ -52,6 +53,11 @@ func updateSettingsUsingFlags(set *CollectorSettings, flags *flag.FlagSet) error
 	if len(resolverSet.URIs) == 0 {
 		return errors.New("at least one config flag must be provided")
 	}
+
+	if confmap.UseUnifiedEnvVarExpansionRules.IsEnabled() && set.ConfigProviderSettings.ResolverSettings.DefaultScheme == "" {
+		set.ConfigProviderSettings.ResolverSettings.DefaultScheme = "env"
```

Comment:
This solution has a downside: it expects that the `ProviderFactories` list includes a provider with the `env` scheme. 

This will be true for all our own usage of `NewCommand`, but may not be true for custom usages.

Reply:
Since this is only affects users who enable the feature gate, and the feature gate is only useful if you're using the env provider, I believe we're safe.